corehq/apps/data_interfaces/signals.py

```diff
@@ -0,0 +1,23 @@
+from casexml.apps.case.models import CommCareCase
+from casexml.apps.case.signals import case_post_save
+
+from corehq.apps.data_interfaces.models import AUTO_UPDATE_XMLNS
+from corehq.form_processor.interfaces.dbaccessors import FormAccessors
+from corehq.toggles import AUTO_CASE_UPDATE_ENHANCEMENTS
+from corehq.form_processor.models import CommCareCaseSQL
+from corehq.form_processor.signals import sql_case_post_save
+
+
+def case_changed_receiver(sender, case, **kwargs):
+    """
+    Spawns a task to run auto update rules tied to the given case.
+    """
+    from corehq.apps.data_interfaces.tasks import run_case_update_rules_on_save
+
+    if FormAccessors(case.domain).get_form(case.xform_ids[-1]).xmlns != AUTO_UPDATE_XMLNS \
```

Comment:
it would be better if the first check we did was the feature flag check, so that we can short circuit most of the time